File: LokynexHealth/backend/src/LokynexHealth.Domain/Entities/Branch.cs
using LokynexHealth.Domain.Enums;

namespace LokynexHealth.Domain.Entities;

public class Branch : BaseEntity
{
    public string BranchName { get; set; } = default!;
    public string BranchCode { get; set; } = default!;
    public string? BranchAddress { get; set; }
    public string? BranchPincode { get; set; }
    public string? BranchPhone { get; set; }
    public string? BranchEmail { get; set; }
    public bool CreatedBySuperAdmin { get; set; }
    public RecordStatus Status { get; set; } = RecordStatus.Active;
    public Guid? CreatedBy { get; set; }

    // Navigation — ekটা Branch e onek User thakte pare
    public ICollection<User> Users { get; set; } = new List<User>();
}

File: LokynexHealth/frontend/src/types/branch.ts
export interface BranchDto {
  id: string;
  branchName: string;
  branchCode: string;
  branchAddress: string | null;
  branchPincode: string | null;
  branchPhone: string | null;
  branchEmail: string | null;
  createdBySuperAdmin: boolean;
  status: string;
  createdAt: string;
}

export interface UpdateBranchRequest {
  branchName: string;
  branchAddress?: string;
  branchPincode?: string;
  branchPhone?: string;
  branchEmail?: string;
  status: string;
}
